src/librustc_data_structures/indexed_set.rs

```diff
@@ -130,6 +130,11 @@ impl<T: Idx> IdxSetBuf<T> {
     pub fn new_empty(universe_size: usize) -> Self {
         Self::new(0, universe_size)
     }
+
+    /// Overwrite one `IdxSetBuf` with another of the same length.
+    pub fn overwrite(&mut self, other: &IdxSetBuf<T>) {
+        self.bits.clone_from_slice(&other.bits);
```

Comment:
This method probably belongs on `IdxSet` rather than `IdxSetBuf` (note that `IdxSetBuf` derefs to `IdxSet`). The relationship between those two types is analogous to `[]` and `Vec`, so most operations that don't change the size or require allocation belong on `IdxSet`. There is however an existing `clone_from` method on that type that does the same thing:

https://doc.rust-lang.org/nightly/nightly-rustc/rustc_data_structures/indexed_set/struct.IdxSet.html#method.clone_from

But maybe we should rename that method (`IdxSet::clone_from`) to `overwrite`? I wonder if we are having shadowing problems because of `Clone::clone_from`....

Reply:
You are right about the shadowing. Here's the DHAT output:
```
==60110== -------------------- 1 of 500 --------------------
==60110== max-live:    217,728 in 1,701 blocks
==60110== tot-alloc:   77,545,592 in 1,330,065 blocks (avg size 58.30)
==60110== deaths:      1,330,065, at avg age 2,374,592 (0.00% of prog lifetime)
==60110== acc-ratios:  0.07 rd, 1.51 wr  (5,953,680 b-read, 117,537,792 b-written)
==60110==    at 0x4C2DECF: malloc (in /usr/lib/valgrind/vgpreload_exp-dhat-amd64-linux.so)
==60110==    by 0x7C016B0: alloc (alloc.rs:62)
==60110==    by 0x7C016B0: alloc (alloc.rs:123)
==60110==    by 0x7C016B0: allocate_in<usize,alloc::alloc::Global> (raw_vec.rs:103)
==60110==    by 0x7C016B0: with_capacity<usize> (raw_vec.rs:147)
==60110==    by 0x7C016B0: with_capacity<usize> (vec.rs:362)
==60110==    by 0x7C016B0: to_vec<usize> (slice.rs:167)
==60110==    by 0x7C016B0: to_vec<usize> (slice.rs:369)
==60110==    by 0x7C016B0: clone<usize> (vec.rs:1670)
==60110==    by 0x7C016B0: clone<rustc::mir::Local> (indexed_set.rs:36)
==60110==    by 0x7C016B0: clone_from<rustc_data_structures::indexed_set::IdxSetBuf<rustc::mir::Local>> (clone.rs:131)
==60110==    by 0x7C016B0: rustc_mir::util::liveness::liveness_of_locals (liveness.rs:144)
==60110==    by 0x7B22D4B: compute (liveness.rs:106)
==60110==    by 0x7B22D4B: rustc_mir::borrow_check::nll::compute_regions (mod.rs:101)
==60110==    by 0x7B33381: rustc_mir::borrow_check::do_mir_borrowck (mod.rs:238)
```
`IdxSetBuf` implements `Clone`, and therefore defines `clone_from`, which is a provided method that calls `clone`. `IdxSet` has a `clone_from` method, which is probably intended to be called, but the `IdxSetBuf` implementation is called instead.

I think renaming `IdxSet::clone_from` as `IdxSet::overwrite` is reasonable. I wonder if there are any other places like this, where the `clone_from` being called is not the one intended.

Reply:
There aren't many definitions of `clone_from`, so it wouldn't be hard to audit.